addons/escoria-core/game/core-scripts/esc/esc_compiler.gd

```diff
@@ -219,8 +219,9 @@ func _compile(lines: Array) -> Array:
 			returned.append(dialog_option)
 		elif command_regex.search(line):
 			var command = ESCCommand.new(line)
-			escoria.logger.trace("Line is the command %s" % command.name)
-			returned.append(command)
+			if command.is_valid():
```

Comment:
I'm not 100% sure, that this works. `is_valid` runs the conditions of the command that check the game state. Are you sure, that compile time is the right time for this to happen?

Reply:
Okay. As discussed on Discord, this is working indeed.